cloud_blobstore/s3.py

```diff
@@ -404,6 +405,55 @@ def get_size(
                 raise BlobNotFoundError(f"Could not find s3://{bucket}/{key}") from ex
             raise BlobStoreUnknownError(ex)
 
+    def multipart_parallel_upload(
```

Comment:
The biggest design concern I have with this is that if you eventually support a GCP multipart upload, this API is not particularly generalized.  It might be worth studying what parameters need to be passed in for GCP multipart upload, and moving the common parameters up to the front, and the S3-specific ones to the end.